### PR TITLE
Add cirq.testing.assert_same_circuits method

### PR DESCRIPTION
**Pull request by Strilanc**

### CODE REVIEW

File: cirq/testing/circuit_compare.py [hunk 1]
```diff
@@ -165,6 +165,48 @@ def assert_circuits_with_terminal_measurements_are_equivalent(
...
+                         expected)
+
+
+def _first_differing_moment_index(circuit1: circuits.Circuit,
+                                  circuit2: circuits.Circuit) -> Optional[int]:
+    differences = [
+        i
+        for i, (m1, m2) in enumerate(itertools.zip_longest(circuit1, circuit2))
+        if m1 != m2
+    ]
```
Comment (dabacon) [inline cirq/testing/circuit_compare.py]: Why not simply bail early for loop?
```
for i, (m1, m2) in enumerate(itertools.zip_longest(circuit1, circuit 2):
  if m1 != m2:
     return i
return None
```

Comment (Strilanc) [reply cirq/testing/circuit_compare.py]: Good point. Done.

**Comment by Strilanc** [comment]
@dabacon PTAL

File: cirq/testing/circuit_compare.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Tuple, Sequence, TYPE_CHECKING
+from typing import Tuple, Sequence, TYPE_CHECKING, Optional
```
Comment (dabacon) [inline cirq/testing/circuit_compare.py]: nit alphabetical?

Comment (Strilanc) [reply cirq/testing/circuit_compare.py]: Acknowledged. I think this is a bit of a lost cause until we add yapf tooling to fix it automatically. It's not consistent anywhere.

**Pull request merged**